### ISSUE TITLE
Primary selection copy paste breaks when input:follow_mouse is set to 2 or 3

### ISSUE THREAD

**Issue by Mrugacz** [hyprwm/Hyprland#8129]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

No

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
$ hyprctl systeminfo -c
Hyprland 0.44.0 built from branch main at commit 01c2ff34ddcb5995409c33c2b549e93b98b56d6b  (layout: simplify the conditions to autogroup (8120)).
Date: Mon Oct 14 20:31:17 2024
Tag: v0.44.0-44-g01c2ff34, commits: 5343
built against aquamarine 0.4.3


flags set:
debug


System Information:
System name: Linux
Node name: desktop
Release: 6.11.3-zen1-1-zen
Version: #1 ZEN SMP PREEMPT_DYNAMIC Thu, 10 Oct 2024 20:11:02 +0000


GPU information: 
0c:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M/6850M XT] [1002:73df] (rev c5) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:

======Config-Start======


Config File: /home/kuba/.config/hypr/hyprland.conf: Read Succeeded
monitor=,preferred,auto,auto

$terminal = kitty
input {
    follow_mouse = 2
}

$mainMod = SUPER # Sets "Windows" key as main modifier

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
bind = $mainMod, RETURN, exec, $terminal
bind = $mainMod, B, exec, firefox
bind = $mainMod, Q, killactive,
bind = $mainMod, P, exit,

======Config-End========

```

</details>

### Description

Setting input:follow_mouse to 2 (or 3) breaks pasting from primary selection, seems to consistently work one way only.
`Firefox -> kitty` works fine
`kitty -> firefox` only works if browser window is opened after selection

Both apps run in wayland mode
```
$ hyprctl clients | grep -i 'window\|class\|xwayland'
Window 64bcad876bb0 -> New Issue · hyprwm/Hyprland — Mozilla Firefox:
	class: firefox
	initialClass: firefox
	xwayland: 0
Window 64bcad59bf10 -> hyprctl clients | grep -i 'window\|class\|xwayland':
	class: kitty
	title: hyprctl clients | grep -i 'window\|class\|xwayland'
	initialClass: kitty
	xwayland: 0
```

### How to reproduce

1. set input:follow_mouse to 2
2. open firefox and kitty 
3. select some text in terminal
4. try to middleclick paste it in firefox

**Comment by hkamphor** [comment]
This is the same issue which I have reported in https://github.com/hyprwm/Hyprland/issues/7352 .
It started when aquamarine was introduced.

The problem occurs with every GTK application (at least GTK 3, not tested with other versions).  When `follow_mouse=2` select text to copy to primary clipboard. Open a GTK application like firefox and it is possible to paste it there. Every new selection copied to the primary clipboard, while firefox is still running, cannot be pasted in firefox. Same for all GTK 3 applications. When the selection was made before starting the GTK 3 application, pasting works, but not the selected/copied text while the GTK 3 application is already running.

This is behavior occurs between:
- GTK3 applications to other GTK3 applications
- Non-GTK applications to GTK3 applications.

This is behavior does not occur between:
- A single GTK3 application window. Inside firefox own window copy/paste works. 
- GTK3 to non-GTK (Testen with firefox to wezterm (wayland))

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!

**Comment by gsgleason** [comment]
This is still broken as of hyprland 0.54.3-4 (arch linux)

Created discussion 14288

https://github.com/hyprwm/Hyprland/discussions/14288
